applications/HDF5Application/CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_model_part_io.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_properties_io.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_data_value_container_io.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_nodal_solution_step_variables_io.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_nodal_solution_step_data_io.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_utilities/hdf5_points_data.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_utilities/hdf5_connectivities_data.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_python/hdf5_python_application.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_io_to_python.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/tests/test_hdf5_file.cpp  
+	${CMAKE_CURRENT_SOURCE_DIR}/tests/test_hdf5_points_data.cpp  
+	${CMAKE_CURRENT_SOURCE_DIR}/tests/test_hdf5_connectivities_data.cpp  
+	${CMAKE_CURRENT_SOURCE_DIR}/tests/test_hdf5_model_part_io.cpp  
+	${CMAKE_CURRENT_SOURCE_DIR}/tests/test_hdf5_nodal_solution_step_data_io.cpp  
+)
+
+set(HDF5_USE_STATIC_LIBRARIES OFF)
+if(MPI_NEEDED)
+  message(STATUS "Searching for parallel HDF5 libraries.")
+  set(HDF5_PREFER_PARALLEL ON)
+  set(KRATOS_HDF5_APPLICATION_SOURCES ${KRATOS_HDF5_APPLICATION_SOURCES} ${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_file_parallel.cpp)
+  set(KRATOS_HDF5_APPLICATION_SOURCES ${KRATOS_HDF5_APPLICATION_SOURCES} ${CMAKE_CURRENT_SOURCE_DIR}/custom_io/hdf5_partitioned_model_part_io.cpp)
+else()
+  message(STATUS "Searching for serial HDF5 libraries.")
+endif()
+
+find_package(HDF5 REQUIRED COMPONENTS C)
+
+if(HDF5_VERSION VERSION_LESS "1.8" )
+  message(FATAL_ERROR "HDF5 1.8 is required.")
+endif()
+
+set(KRATOS_HDF5_APPLICATION_EXTRA_LIBS "")
+
+if(MPI_NEEDED)
+  if(HDF5_IS_PARALLEL)
+    message(STATUS "Using parallel HDF5 libraries.")
+    if (   (HDF5_INCLUDE_DIRS MATCHES ".*openmpi.*")
+        OR (HDF5_LIBRARY_DIRS MATCHES ".*openmpi.*"))
+      add_definitions(-DOMPI_SKIP_MPICXX)
+    endif()
+    set(KRATOS_HDF5_APPLICATION_EXTRA_LIBS ${KRATOS_HDF5_APPLICATION_EXTRA_LIBS} ${MPI_LIBRARIES})
+    ##################################################################
+    # If cmake configures another application using HDF5 in serial,
+    # the find_library() used in FindHDF5.cmake may find the serial
+    # version of libhdf5.so even though the directory passed with
+    # HINTS is for the parallel version. If the other application
+    # cannot be switched off, a workaround is to call find_library
+    # with NO_CMAKE_PATH:
+    #    find_library(HDF5_LIBRARY
+    #                 NAMES "hdf5"
+    #                 HINTS ${HDF5_C_LIBRARY_DIRS}
+    #                 ENV HDF5_ROOT
+    #                 PATH_SUFFIXES lib Lib 
+    #                 NO_CMAKE_PATH)
+    ##################################################################
+  else()
+    message(FATAL_ERROR "Parallel HDF5 libraries were not found.")
+  endif()
+else()
+  if(HDF5_IS_PARALLEL)
+    message(FATAL_ERROR "Parallel HDF5 libraries were found.")
```

Comment:
I encountered this message while trying to compile the application and I found it slightly confusing... Maybe specify that this is a non-parallel compilation, so it needs a non-parallel hdf5 version?

Reply:
fixed.